ibm/mas_devops/roles/db2/tasks/install/main.yml

```diff
@@ -0,0 +1,413 @@
+
+# Look up the default channel for the db2u-operator package manifest
+- name: Lookup db2u-operator packagemanifest
+  kubernetes.core.k8s_info:
+    api_version: packages.operators.coreos.com/v1
+    kind: PackageManifest
+    name: db2u-operator
+    namespace: ibm-common-services
+  register: db2u_manifest_info
+  until: db2u_manifest_info.resources[0].status.defaultChannel is defined
+  retries: 60 # Approximately 30 minutes before we give up
+  delay: 30 # 30 seconds
+  when: db2_channel is not defined or db2_channel == "" or
+        db2_version is not defined or db2_version == ""
+
+- name: Set db2u-operator channel
+  ansible.builtin.set_fact:
+    db2_channel: "{{ db2u_manifest_info.resources[0].status.defaultChannel }}"
+  when: db2_channel is not defined or db2_channel == ""
+
+- name: Set db2 version
+  ansible.builtin.set_fact:
+    db2_version: "{{ db2u_manifest_info.resources[0].status.channels | selectattr('name', 'equalto', db2_channel) | map(attribute='currentCSVDesc') | map(attribute='annotations') | map(attribute='productVersion') | first }}"
+  when: db2_version is not defined or db2_version == ""
+
+
+# Fail if required parameters are not set
+# -----------------------------------------------------------------------------
+- name: "Verify db2_channel and db2_version set"
+  assert:
+    that:
+      - db2_channel is defined and db2_channel != ""
+      - db2_version is defined and db2_version != ""
+    fail_msg: "Unable to determine db2_channel and/or db2_version from catalog"
+
+
+- name: Debug Db2 Universal Operator Install
+  ansible.builtin.debug:
+    msg:
+      - "Db2 Channel ............................ {{ db2_channel }}"
+      - "Db2 Version ............................ {{ db2_version }}"
+
+
+# 5. Install a Db2u Operator
+# -----------------------------------------------------------------------------
+- name: "Create db2u Namespace"
+  kubernetes.core.k8s:
+    apply: yes
+    definition: "{{ lookup('template', 'templates/db2u_namespace.yaml') }}"
+  register: _db2_namespace_result
+
+- name: Set 'ibm-registry' secret content
+  vars:
+    entitledAuthStr: "{{ registry_user }}:{{ ibm_entitlement_key }}"
+    entitledAuth: "{{ entitledAuthStr | b64encode }}"
+    content:
+      - "{\"auths\":{\"{{ registry }}/cp/cpd\":{\"username\":\"{{ registry_user }}\",\"password\":\"{{ ibm_entitlement_key }}\",\"email\":\"{{ registry_user }}\",\"auth\":\"{{ entitledAuth }}\"}"
+      - "}"
+      - "}"
+  set_fact:
+    new_secret: "{{ content | join('') }}"
+
+- name: "Generate 'ibm-registry' secret"
+  kubernetes.core.k8s:
+    definition:
+      apiVersion: v1
+      kind: Secret
+      type: kubernetes.io/dockerconfigjson
+      metadata:
+        name: ibm-registry
+        namespace: "{{ db2_namespace }}"
+      data:
+        .dockerconfigjson: "{{ new_secret | to_json | b64encode }}"
+  register: secretUpdateResult
+
+- name: "Create Db2 Universal Operator Operand Request"
```

Comment:
Can we fix the name of this "Delete old OperandRequest" .. it's very confusing that it still says "Create", when it's actually now just a cleanup step.

Reply:
Changed